kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/KilkariServiceImpl.java

```diff
@@ -20,12 +23,22 @@
 
     @Override
     public List<SubscriptionPack> getSubscriberPacks(String callingNumber) {
+        Set<Subscription> subscriptions = getSubscriptions(callingNumber);
+        List<SubscriptionPack> subscriptionPacksForSubscriber = new ArrayList<>();
+
+        for (Subscription subscription : subscriptions) {
+            subscriptionPacksForSubscriber.add(subscription.getSubscriptionPack());
+        }
+        return subscriptionPacksForSubscriber;
+    }
+
+    @Override
+    public Set<Subscription> getSubscriptions(String callingNumber) {
         Subscriber subscriber = subscriberDataService.findByCallingNumber(callingNumber);
         if (null == subscriber) {
             //todo: handle that better (ie: proper exception?) such that the proper bad response is returned
             throw new IllegalStateException("NULL subscriber");
```

Comment:
I think that makes the most sense.  Doing it in here means we have to
handle the null case (and clearly we don't know how to).  Also all this
function really does is call subscriber.getSubscriptions() and the calling
code can do that themselves.